File: android/app/src/test/java/com/thankful/app/VerifierTests.kt
package com.thankful.app

import com.thankful.app.core.ApiClient
import com.thankful.app.core.VerifierService
import com.thankful.app.data.CachedTicket
import com.thankful.app.data.TicketsCacheService
import org.junit.Assert.assertThrows
import org.junit.Assert.assertTrue
import org.junit.Test
import org.mockito.Mockito
import android.content.Context

class VerifierTests {
    @Test
    fun verifyOfflineRejectsBadFormat() {
        val api = ApiClient("https://example.invalid")
        val verifier = VerifierService(api)
        assertThrows(Exception::class.java) { verifier.verifyOffline("bad.token") }
    }

    @Test
    fun ticketsCacheRoundtrip() {
        val ctx = Mockito.mock(Context::class.java)
        // For EncryptedSharedPreferences, a real Android test environment is needed; this is a placeholder
        val cache = TicketsCacheService(ctx)
        val org = "org_test"
        val t = CachedTicket("j1", "tkn", "ord", "p1", "A-1", org, 0, null)
        cache.upsert(org, t)
        val list = cache.list(org)
        assertTrue(list.any { it.jti == "j1" })
        cache.remove(org, "j1")
    }
}

File: ios/ThankfulAppTests/VerifierTests.swift
import XCTest
@testable import ThankfulApp

final class VerifierTests: XCTestCase {
  func testVerifyOfflineRejectsBadFormat() {
    let api = ApiClient(baseURL: URL(string: "https://example.invalid")!)
    let verifier = VerifierService(api: api)
    XCTAssertThrowsError(try verifier.verifyOffline(token: "bad.token"))
  }

  func testTicketsCacheRoundtrip() {
    let cache = TicketsCacheService()
    let org = "org_test"
    let ticket = CachedTicket(jti: "j1", token: "tkn", orderId: "ord1", performanceId: "p1", seatId: "A-1", tenantId: org, issuedAt: 0, expiresAt: nil)
    cache.upsert(orgId: org, ticket: ticket)
    let list = cache.list(orgId: org)
    XCTAssertTrue(list.contains(where: { $0.jti == "j1" }))
    cache.remove(orgId: org, jti: "j1")
    XCTAssertFalse(cache.list(orgId: org).contains(where: { $0.jti == "j1" }))
  }
}


